server/fleet/windows_mdm.go

```diff
@@ -29,6 +36,57 @@ type MDMWindowsConfigProfile struct {
 	UpdatedAt   time.Time `db:"updated_at" json:"updated_at"`
 }
 
+// ValidateUserProvided ensures that the SyncML content in the profile is valid
+// for Windows.
+//
+// It checks that all top-level elements are <Replace> and none of the <LocURI>
+// elements within <Target> are reserved URIs.
+//
+// Returns an error if these conditions are not met.
+func (m *MDMWindowsConfigProfile) ValidateUserProvided() error {
+	if mdm.GetRawProfilePlatform(m.SyncML) != "windows" {
+		return errors.New("Only <Replace> supported as a top level element. Make sure you don’t have other top level elements.")
+	}
+
+	doc := etree.NewDocument()
+	if err := doc.ReadFromBytes(m.SyncML); err != nil {
+		return fmt.Errorf("Couldn’t upload. The file should include valid XML: %w", err)
+	}
+
+	for _, element := range doc.ChildElements() {
+		if element.Tag != CmdReplace {
+			return errors.New("Only <Replace> supported as a top level element. Make sure you don’t have other top level elements.")
+		}
+
+		for _, target := range element.FindElements("Target") {
+			locURI := target.FindElement("LocURI")
+			if locURI != nil {
+				if err := validateFleetProvidedLocURI(locURI.Text()); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+var fleetProvidedLocURIValidationMap = map[string][2]string{
+	microsoft_mdm.FleetBitLockerTargetLocURI: {"BitLocker", "mdm.enable_disk_encryption"},
+	microsoft_mdm.FleetOSUpdateTargetLocURI:  {"Windows updates", "mdm.windows_updates"},
+}
+
+func validateFleetProvidedLocURI(locURI string) error {
+	sanitizedLocURI := strings.TrimSpace(locURI)
+	for fleetLocURI, errHints := range fleetProvidedLocURIValidationMap {
+		if strings.Contains(sanitizedLocURI, fleetLocURI) {
+			return fmt.Errorf("Custom configuration profiles can’t include %s settings. To control these settings, use the %s option.", errHints[0], errHints[1])
```

Comment:
Hey @roperzh , much appreciated and that's a very elegant solution! I think for the UI we'll want "go to {Disk encryption,OS updated}" instead of "use the ... option" and (the more tricky thing) I believe it's supposed to be a link when displayed by the frontend. But no worries for this PR, I'll do the required changes as necessary, it's awesome that you've put the mechanism in place!

Reply:
ahhh, good point! this might be a bad abstraction in that case, sorry about that!

Reply:
not necessarily to do in this PR, but maybe this should just return different errors, or an error with a method to get the location uri that failed and the consumer upstream defines the actual message?

Reply:
Yeah we can figure out a way to handle this later on as you say, feel free to merge as-is!